source/main/gui/panels/GUI_GameConsole.cpp

```diff
@@ -235,183 +236,219 @@ void Console::eventCommandAccept(MyGUI::Edit* _sender)
+        Real h = gEnv->terrainManager->getHeightFinder()->getHeightAt(pos.x, pos.z);
+        putMessage(CONSOLE_MSGTYPE_INFO, CONSOLE_SYSTEM_REPLY, _L("Terrain height at position: ") + String("x: ") + TOSTRING(pos.x) + String("z: ") + TOSTRING(pos.z) + _L(" = ") + TOSTRING(h), "world.png");
 
-            StringUtil::trim(command);
-            if (command.empty())
-                return;
+        return;
+    }
+    else if (args[0] == "ver")
+    {
+        putMessage(CONSOLE_MSGTYPE_INFO, CONSOLE_TITLE, "Rigs of Rods:", "information.png");
+        putMessage(CONSOLE_MSGTYPE_INFO, CONSOLE_SYSTEM_REPLY, " Version: " + String(ROR_VERSION_STRING), "information.png");
+        putMessage(CONSOLE_MSGTYPE_INFO, CONSOLE_SYSTEM_REPLY, " Protocol version: " + String(RORNET_VERSION), "information.png");
+        putMessage(CONSOLE_MSGTYPE_INFO, CONSOLE_SYSTEM_REPLY, " build time: " + String(ROR_BUILD_DATE) + ", " + String(ROR_BUILD_TIME), "information.png");
+        return;
+    }
+    else if (args[0] == "quit")
+    {
+        RoR::App::app_state.SetPending (RoR::AppState::SHUTDOWN);
+        return;
+    }
+#ifdef USE_ANGELSCRIPT
+    else if (args[0] == "as" && (is_appstate_sim && !is_sim_select))
+    {
+        // we want to notify any running scripts that we might change something (prevent cheating)
+        ScriptEngine::getSingleton().triggerEvent(SE_ANGELSCRIPT_MANIPULATIONS);
+
+        String command = msg.substr(args[0].length());
 
-            String nmsg = RoR::Color::ScriptCommandColour + ">>> " + RoR::Color::NormalColour + command;
-            putMessage(CONSOLE_MSGTYPE_SCRIPT, CONSOLE_LOCAL_SCRIPT, nmsg, "script_go.png");
-            int res = ScriptEngine::getSingleton().executeString(command);
+        StringUtil::trim(command);
+        if (command.empty())
             return;
-        }
+
+        String nmsg = RoR::Color::ScriptCommandColour + ">>> " + RoR::Color::NormalColour + command;
+        putMessage(CONSOLE_MSGTYPE_SCRIPT, CONSOLE_LOCAL_SCRIPT, nmsg, "script_go.png");
+        int res = ScriptEngine::getSingleton().executeString(command);
+        return;
+    }
 #endif //ANGELSCRIPT
-        else if (args[0] == "log")
+    else if (args[0] == "log")
+    {
+        // switch to console logging
+        bool logging = BSETTING("Enable Ingame Console", false);
+        if (!logging)
         {
-            // switch to console logging
-            bool now_logging = !App::diag_log_console_echo.GetActive();
-            const char* msg = (now_logging) ? " logging to console enabled" : " logging to console disabled";
-            this->putMessage(CONSOLE_MSGTYPE_INFO, CONSOLE_SYSTEM_NOTICE, _L(msg), "information.png");
-            App::diag_log_console_echo.SetActive(now_logging);
-            return;
+            putMessage(CONSOLE_MSGTYPE_INFO, CONSOLE_SYSTEM_NOTICE, _L(" logging to console enabled"), "information.png");
+            SETTINGS.setSetting("Enable Ingame Console", "Yes");
```

Comment:
BSETTING won't work anymore because "Enable ingame console" is now transformed to GVar: https://github.com/RigsOfRods/rigs-of-rods/blob/025ff0a335b3be6afde16da677870f42506eb1fa/source/main/Application.cpp#L100 and thus it's deleted from the `Settings` hashmap: https://github.com/RigsOfRods/rigs-of-rods/blob/025ff0a335b3be6afde16da677870f42506eb1fa/source/main/utils/Settings.cpp#L823-L829

I'm thinking you just grabbed your old modified file and pasted it over this one, didn't you?